adminSiteClient/GrapherConfigGridEditor.tsx

```diff
@@ -0,0 +1,1267 @@
+            )
+            return
+        }
+
+        // Since we are editing only values in one column, we strongly assume
+        // that all values are of the same type (actually supposed to be the same value)
+        // -> verify this
+        const allChangesSameType = changes.every(
+            (change) =>
+                typeof change[newValueIndex] ===
+                typeof changes[0][newValueIndex]
+        )
+        if (!allChangesSameType) {
+            console.error("Not all changes were of the same type?")
+            return
+        }
+
+        // Check if the type matches the target (e.g. to avoid a string value ending
+        // up in a text field). This is hard to nail down fully but we try to avoid
+        // such mistakes where possible.
+        const firstNewValue = changes[0][newValueIndex]
+        const invalidTypeAssignment = match<
+            [FieldType | FieldType[], string],
+            boolean
+        >([columnDefinition.type, typeof firstNewValue])
+            .with([FieldType.string, "string"], () => false)
+            .with([FieldType.number, "string"], () =>
+                Number.isNaN(Number.parseFloat(firstNewValue))
+            )
+            .with([FieldType.integer, "string"], () =>
+                Number.isNaN(Number.parseInt(firstNewValue))
+            )
+            .with([FieldType.boolean, "boolean"], () => false)
+            .with([[__], "string"], () => false)
+
+            .otherwise(() => true)
+        if (invalidTypeAssignment) return false
+
+        // TODO: locally apply values here and see if they parse the schema? Might be an easy
+        // highlevel way to verify that enums are obeyed etc. But might also be tricky to debug
+        // if validation fails for nonobvious reasons
+
+        // If we got here then everything seems to be ok
+        return
+    }
+
+    processChangedCells(
+        changes: Handsontable.CellChange[] | null,
+        source: Handsontable.ChangeSource
+    ): void {
+        // We don't want to persist changes due to loading data
+        const { fieldDescriptionsMap, richDataRows } = this
+        if (
+            source === "loadData" ||
+            changes === null ||
+            richDataRows === undefined
+        )
+            return
+
+        const patches = []
+        // Collect the patches, which is a data structure of row, column, oldValue, newValue
+        for (const change of changes) {
+            let [rowIndex, column, prevVal, newVal] = change
+
+            const fieldDesc = fieldDescriptionsMap.get(column as string)
+            const row = richDataRows[rowIndex]
+
+            // we normalize the default values to null (we don't want to store them and null leads to removing the key)
+            if (
+                fieldDesc?.default !== undefined &&
+                prevVal === fieldDesc?.default
+            )
+                prevVal = null
```

Comment:
I know this is not new code, but I just realized that it goes wrong.
Try to change `/map/colorScale/colorSchemeInvert` using the bulk chart editor, and it will throw `When trying to set value for 364 at /map/colorScale/colorSchemeInvert, the existing value was false instead of null`.

What's happening here is that the persisted value for `map.colorScale.colorSchemeInvert` is `false`, but we're wrongly "normalizing" it to `null` in this line.
An example chart where this happens is 364.

Reply:
Good catch. Fixed this with a new boolean field on patch that communicates if the old value should be seen as equivalent to null or undefined.

Reply:
Hm, does the fix need to be this complicated? Can we not simply skip the normalization step for `prevVal`?

Reply:
I don't think it can be simplified. We have two values that are equivalent: the default value and the absence. When we encounter either we are fine and proceed (this includes on the server where in between there might have been a change).